### DISCUSSION TITLE
PIMCORE_ASSETS_DIRECTORY not usable anymore ?

### DISCUSSION THREAD

**Discussion by lalmat** [pimcore/pimcore#9879] [Q&A / Support] [answered]
Hi everyone,

According to the documentation and the "config/pimcore/constants.example.php" (in pimcore/skeleton) we can override configuration constants. After installing Pimcore X, I can't override the PIMCORE_ASSETS_DIRECTORY constant. I'm searching why...

After looking at the source of lib/Bootstrap.php that manage this, it seems that the PIMCORE_ASSETS_DIRECTORY constant is not defined anymore so I guess that why it's not overriden. 

So... Is this a bug or is there some undocumented way to achive the asset's directory override ?

Thanks.

**Comment by NiklasBr** [comment]
The constant was named `PIMCORE_ASSET_DIRECTORY` and was removed ([deprecated in 6.9.0](https://pimcore.com/docs/pimcore/6.9/Development_Documentation/Installation_and_Upgrade/Upgrade_Notes/index.html#page_6-9-0)). The docs and comments lacks any information *why* it was deprecated and removed though. So it's up to us to guess why Pimcore chose to do it, and it is also up to us to guess what to do instead. There is certainly a perfectly valid reason, but what was the reason?

[The commit](https://github.com/pimcore/pimcore/commit/d28adbe0e621b892fdb4c89284a2dfde4673898b).

**Reply by fashxp** [reply]
The reason is that we now use flysystem for storing assets. And thus, there is no absolute file path anymore...

**Reply by lalmat** [reply]
Yes I know I've read that, but there no help here in the doc : https://pimcore.com/docs/pimcore/10.1/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html

on "how" to override the default configuration file inside the pimcore/skeleton boilerplate ?

The pimcore/skeleton is in version 10.X but there is always some references to the old static env variables inside...

I'm very confused.

**Reply by lalmat** [reply]
Ok I found !

I've created a flysystem.yaml file in "/config/packages/prod" directory with the content of the template from the documentation.

Then I've imported this into the config.yaml
```
imports:
   (...)
    - { resource: flysystem.yaml }
```

Thanks.

**Comment by lalmat** [comment, marked as answer]
Ok I found !

I've created a `flysystem.yaml` file in "/config/packages/prod" directory with the content of the template from the documentation here:
https://pimcore.com/docs/pimcore/10.1/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html

Then I've imported this into the `config.yaml`
```
imports:
   (...)
    - { resource: flysystem.yaml }
 ```
Thanks.
